numba/inline_closurecall.py

```diff
@@ -420,12 +398,60 @@ def inline_closure_call(func_ir, glbls, block, i, callee, typingctx=None,
             work_list.append(block)
     return callee_blocks, var_dict
 
+
+def _get_callee_args(call_expr, callee, loc, func_ir):
+    """Get arguments for calling 'callee', including the default arguments.
+    keyword arguments are currently only handled when 'callee' is a function.
+    """
+    args = list(call_expr.args)
+    debug_print = _make_debug_print("inline_closure_call default handling")
+
+    # handle defaults and kw arguments using pysignature if callee is function
+    if isinstance(callee, pytypes.FunctionType):
+        pysig = numba.utils.pysignature(callee)
+        normal_handler = lambda index, param, default: default
+        default_handler = lambda index, param, default: ir.Const(default, loc)
+        # TODO: handle stararg
```

Comment:
done.

Reply:
Unfortunately I don't think what is implemented will work as this line is invalid because `obj` is not defined:
https://github.com/numba/numba/pull/4536/files#diff-69e01738577449bec5a17f54f5845f47R417

further there's no test case written for this eventuality.

Reply:
Thanks. Coding while asleep doesn't work :) should be fixed now.